[IMP] website/ecommerce: update order handling page (#15985)
odoo/documentation: content/applications/websites/ecommerce/order_handling.rst

Order handling
==============

In Odoo's e-commerce workflow, an online purchase typically goes through three main steps:
:ref:`sale <ecommerce/handling/sales>`, :ref:`delivery <ecommerce/handling/delivery>`, and
:ref:`invoicing <ecommerce/handling/invoices>`. Each step generates key documents that ensure
smooth order processing from start to finish, including sales orders upon order confirmation,

> delivery orders to manage :doc:`picking

Reviewer: ```suggestion
:ref:`invoicing <ecommerce/handling/invoices>`. Each step generates key documents, including sales orders upon order confirmation, delivery orders to manage :doc:`picking
```
(I think this part of the sentence is a little bit too marketing-oriented and doesn't really add much information-wise 😉 )

Sale
====

Once a customer adds a product to their cart, a quotation is automatically created in the backend.
To view all quotations and orders, navigate to :menuselection:`Website --> eCommerce -->
Orders`, and select one to open it. The status bar at the top of the order shows its current status:

- :guilabel:`Quotation`: A product has been :ref:`added to the cart
  <ecommerce/checkout/add-to-cart>`, but the customer has not yet completed the :ref:`checkout
  process <ecommerce/checkout/steps>`. If the customer is not logged in, the quotation is assigned
  to the default *Public User* contact.
- :guilabel:`Quotation Sent`: The customer has completed the :ref:`checkout process
  <ecommerce/checkout/steps>` and confirmed the order, but the :ref:`payment
  <ecommerce/checkout/payment>` has not yet been confirmed.
- :guilabel:`Sales Order`: The customer has completed the checkout process, confirmed the order,
  and the payment has been successfully received.

.. note::
   This three-step process is not visible to customers. However, they can view their orders and
   check their status through the customer :doc:`portal </applications/general/users/user_portals>`.

.. tip::

>    - The :guilabel:`From Website` and :guilabel:`Confirmed` filters are enabled by default in the

Reviewer: I'm not sure I would mention the "From website" filter here, I don't think it's relevant in this case because in the context of e-commerce, I don't think people would want to see non-e-commerce orders (and we're going to assume that's out of the ecommerce scope anyways documentation-wise); what's really important to mention is the Confirmed filter so I'd just keep that one 🙂

     :guilabel:`Orders` list. To view *all* records, including quotations, remove the filters by

>      clicking the :icon:`fa-times` :guilabel:`(Remove)` icon.

Reviewer: I think technically a quotation is a draft sales order, and since the list view is named "Orders", I guess that could also justify it? If you prefer, you could also just rephrase the sentence to say: *to view all e-commerce orders and quotations, remove the filter.*

   - To view only orders in the :guilabel:`Quotation Sent` stage, go to :menuselection:`Website -->
     eCommerce --> Unpaid Orders`.
   - E-commerce orders can be automatically assigned to a specific sales team or salesperson. To do
     so, go to :menuselection:`Website --> Configuration --> Settings`. In the :guilabel:`eCommerce`
     section, under :guilabel:`Orders Assignment`, select a :guilabel:`Sales Team` or
     :guilabel:`Salesperson` to handle e-commerce orders.

.. seealso::
   - :doc:`checkout`
   - :doc:`/applications/finance/payment_providers`

.. _ecommerce/handling/abandoned_cart:

Abandoned cart
--------------

An abandoned cart represents an order for which the customer did not complete the :ref:`checkout

> process <ecommerce/checkout/steps>`. It is possible to automatically send an email reminder to

Reviewer: See my comment on line 79

the customer for these orders.

To enable abandoned cart reminders:

#. Go to :menuselection:`Website --> Configuration --> Settings`.
#. In the :guilabel:`eCommerce` section, enable :guilabel:`Follow up abandoned carts`.
#. Set the time delay after which the reminder email is sent in the :guilabel:`Send after` field and
   click :guilabel:`Save`.
#. Click the :icon:`fa-arrow-right` :guilabel:`Customize Abandoned Email Template`
   link to configure an :doc:`email template </applications/general/companies/email_template>`.

To view all abandoned carts, go to :menuselection:`Website --> eCommerce --> Abandoned Carts`.

.. note::
   - To receive reminders, customers must have entered their contact details during

>      checkout or be logged in when adding items to the cart.

Author: They still can add their contact details when starting the checkout process, but not finishing it, which actually means "the customer did not complete the checkout out process".
For your suggestion on this line: I would explicitly mention that the customers must be logged in or add the contact information. TBH I think your suggestion is a bit misleading as it does not mention that there are two ways of adding contact information in order to make abandoned carts reminders possible. I don't know if you understand what I mean :/

>      checkout or be logged in when adding items to the cart.

Author: Sorry, I missed your last comment :) 
That's correct! As long as the setting has been enabled, of course. 
Yes, when we don't have customer details, it will be assigned to a public user and won't appear under "Abandoned carts."
I will think about how to include this clarifying information :) Thank you!

   - Abandoned cart reminders are only sent for carts created after the feature is enabled.
     Existing abandoned carts are not affected.

.. _ecommerce/handling/delivery:

Delivery
========

After confirming an order, a delivery order is automatically created. To access it, click
the :icon:`fa-truck` :guilabel:`Delivery` smart button on the sales order, and click
:guilabel:`Validate` once the order has been shipped.

.. note::
   - The :doc:`Inventory app </applications/inventory_and_mrp/inventory>`

>      must be installed to process deliveries.

Author: As a note? Or as some kind of introduction?

   - The :guilabel:`Delivery` button does not appear on the sales order if the product type is set
     to :guilabel:`Service` or if :guilabel:`Recurring` is selected in the :ref:`product form
     <ecommerce/products/product-form>`.

> Packing e-commerce orders typically involves :doc:`picking

Reviewer: This sounds a bit weird said like this because we're talking about *preparing* orders, and we've already talked about shipping them (*click Validate once the order has been shipped* on line 90). So I think I would move the different info in the Delivery section around a bit and rephrase to reflect the logical order of the operations (picking, packaging, printing labels, shipping THEN validate the delivery order, if that's what people usually do; I'd check this).

----------------

To ensure that products ordered by customers are reserved in stock, select the
:doc:`At Confirmation reservation option
</applications/inventory_and_mrp/inventory/shipping_receiving/reservation_methods/at_confirmation>`
on the :guilabel:`Delivery orders` operation type. When enabled, product quantities are
automatically reserved once the order is confirmed, preventing items from becoming unavailable
after the order is placed.

.. note::
   If customers pay via :ref:`wire transfer <payment_providers/bank_payments/wire_transfer>`,
   the quotation is *not* automatically confirmed, and stock is *not* reserved. In this case,
   the customer receives a notification on the :guilabel:`My account` page of their
   :doc:`customer account <customer_accounts>` to review and pay for the order. Once
   payment is completed, the order must be manually confirmed in the backend to
   reserve products in stock. If the customer :ref:`pays upon in-store
   <payment_providers/inperson_payments/pay_on_site>` :ref:`pickup
   <ecommerce/shipping/instore-pickup>` or :ref:`delivery
   <payment_providers/inperson_payments/cash_on_delivery>`, the order is confirmed and the quantity

>    is reserved as usual.

Reviewer: I think the first part of this note would fit better under the Sale section since we're talking about order confirmation mostly. So I'd suggest moving that part of the note there, and then remove this note here (since what should really be here is the fact that products are only reserved once the order is confirmed and you already said that on line 125).

.. seealso::
   :doc:`shipping`

.. _ecommerce/handling/returns:

Returns and refunds
-------------------

Customers can :doc:`return </applications/sales/sales/products_prices/returns>` products
through their :doc:`portal </applications/general/users/user_portals>` by navigating to the
:guilabel:`My Orders` section, selecting the relevant order, and clicking the :guilabel:`Return`
button. A return document is automatically generated.

.. image:: order_handling/return-button.png

>    :alt: Return button in the portal.

Reviewer: For some reason, in all runbots/DBs I tested, the return button is yellow:

<img width="514" height="118" alt="Image" src="https://github.com/user-attachments/assets/47501d77-e4dd-48b0-a846-235fceec1e5c" />

> Once the returned items are received, initiate the :doc:`return and refunds process

Reviewer: ```suggestion
Once the returned items are received, initiate the :doc:`return and refund process
```

</applications/sales/sales/products_prices/returns>` in the backend.

.. note::
   - Returning products may not be possible depending on the

>      :doc:`return </applications/sales/sales/products_prices/returns>` strategy or product type.

Reviewer: I didn't understand at first why you had the exact same link twice (on line 156 and here) so close to one another, so I suggest changing this slightly to explicitely mention the two strategies:

```suggestion
     return strategy (i.e., :ref:`before <add ref> or :ref:`after invoicing <add ref>`) or product type.
```
but then I wonder if I really understood this correctly because I don't understand how the result strategy as explained in the linked doc has an impact on the possibility to return products? What do you mean exactly by *return strategy* then?

Author: I meant after or before invoicing, but indeed, "strategy" and the whole sentence are a bit misleading. I will rephrase

   - Full :doc:`refunds </applications/sales/sales/products_prices/returns>` can be issued directly
     to customers from within the order interface. First, a refund-compatible
     :ref:`payment provider <payment_providers/supported_providers>` needs to be enabled.

.. tip::
   The :guilabel:`Return` button in the portal is a standard feature and can only be removed in
   :doc:`/applications/general/developer_mode`. Once enabled, go to :menuselection:`Settings -->
   Technical --> Views`. Search for the `Orders Shipping Followup` view, and comment out the
   relevant lines to remove the button.

.. seealso::
   :doc:`/applications/services/helpdesk/advanced/after_sales`

.. _ecommerce/handling/invoices:

Invoicing
=========

The final step of an e-commerce order is generating and sending the :doc:`customer invoice
</applications/finance/accounting/customer_invoices>`. Depending on your needs, an invoice can
either be generated automatically or on demand. This process can be automated if
(and when) the :doc:`online payment </applications/finance/payment_providers>` is :doc:`confirmed

> </applications/sales/sales/sales_quotations/get_paid_to_validate>`.

Reviewer: ```suggestion
</applications/finance/accounting/customer_invoices>`. Depending on your needs, invoices can
be generated manually or automatically.
```
I would then move the part about payments (I'd write payment and not online payment since it's technically not always an online payment, e.g., wire transfer, pay on site, etc.) in a note and *after* the instructions (on line 194). I would also remove the link for payment registration, I don't think the linked doc is super relevant in the context of e-commerce orders. I'd consider adding a seealso with a link to this doc instead: https://www.odoo.com/documentation/master/applications/finance/accounting/payments.html

```suggestion
Invoices are only sent automatically if (and when) the :doc:`payment </applications/finance/payment_providers>` is registered.
```

To automate invoicing:

#. Go to :menuselection:`Website --> Configuration --> Settings`.
#. In the :guilabel:`eCommerce` section, enable :guilabel:`Automatic Invoice`.

> #. Select an :guilabel:`Email Template` to send to the customer.

Reviewer: ```suggestion
#. Select the relevant :guilabel:`Email Template`, then click :guilabel:`Save`.
```

#. Customize the :doc:`email template </applications/general/companies/email_template>` by
   clicking the :icon:`fa-arrow-right` icon next to the template name.

Customers can view their invoices under :guilabel:`Your Invoices` in the
:doc:`portal </applications/general/users/user_portals>`.

.. note::
   Once a customer pays through the e-commerce, Odoo creates an invoice that initially appears as a
   :guilabel:`Proforma Invoice` in the customer portal, even if payment has already been
   completed. The document displays as :guilabel:`Invoice` in the portal only after manually
   clicking :guilabel:`Send & Print` on the invoice form in the backend. You can avoid this behavior

>    by enabling the :guilabel:`Automatic Invoice` setting.

Reviewer: This seems a bit weird here since we just talked about automatic invoicing. I'd rephrase so it makes a bit more sense here:

```suggestion
   When automatic invoicing is disabled and invoices are only created manually, they appear as :doc:`pro-forma invoices <https://www.odoo.com/documentation/19.0/applications/sales/sales/invoicing/proforma.html>` in the customer portal until they are :ref:`sent to the customer <https://www.odoo.com/documentation/master/applications/finance/accounting/customer_invoices.html#invoice-sending>`.
```

Order handling
==============

In Odoo's e-commerce workflow, an online purchase typically goes through three main steps:
:ref:`sale <ecommerce/handling/sales>`, :ref:`delivery <ecommerce/handling/delivery>`, and
:ref:`invoicing <ecommerce/handling/invoices>`. Each step generates key documents, including sales
orders upon order confirmation, delivery orders to manage :doc:`picking
</applications/inventory_and_mrp/inventory/shipping_receiving/picking_methods>`,
:doc:`packing
</applications/inventory_and_mrp/inventory/shipping_receiving/setup_configuration/multipack>`, and
:doc:`shipping <shipping>`, and invoices. Odoo eCommerce also offers additional tools to
streamline sales and delivery processes, such as sending :ref:`abandoned cart emails
<ecommerce/handling/abandoned_cart>`, ensuring :ref:`stock availability
<ecommerce/handling/stock-management>`, and handling product :ref:`returns and refunds
<ecommerce/handling/returns>`.

.. _ecommerce/handling/sales:

Sale
====

> Once a customer adds a product to their cart, a quotation is automatically created in the backend.

Reviewer: I think I would remove this sentence, for the following reasons:

1) this is already explained in the quotation list item (line 27)
2) if we talk about quotations here in this intro, I feel like we should also talk about sales orders and order confirmation (but then again, see point 1), even more so because we say "to view all quotations and order" in the next sentence

  to the default *Public User* contact.
- :guilabel:`Quotation Sent`: The customer has completed the :ref:`checkout process
  <ecommerce/checkout/steps>` and confirmed the order, but the :ref:`payment
  <ecommerce/checkout/payment>` has not yet been confirmed.
- :guilabel:`Sales Order`: The customer has completed the :ref:`checkout process
  <ecommerce/checkout/steps>`, confirmed the order, and the :ref:`payment
  <ecommerce/checkout/payment>` has been successfully received.

.. note::
   - This three-step process is not visible to customers. However, they can view their orders and
     check their status through the customer :doc:`portal
     </applications/general/users/user_portals>`.
   - If the customer pays via :ref:`wire transfer <payment_providers/bank_payments/wire_transfer>`,
     the quotation is *not* automatically confirmed. In this case, the customer receives a
     notification on the :guilabel:`My account` page of their :doc:`customer account
     <customer_accounts>` to review and pay for the order. Once payment is completed, the order
     must be manually confirmed in the backend.

> .. tip::

Reviewer: I would personnally have the tip *before* the note, especially because of the first 2 list items, which I find are really important since people could really wonder why they don't see quotations when they access the Orders list view.

:ref:`checkout process <ecommerce/checkout/steps>` or leaves the website before completing it. It
is possible to automatically send an email reminder to the customer for these orders.

To enable abandoned cart reminders:

#. Go to :menuselection:`Website --> Configuration --> Settings`.
#. In the :guilabel:`eCommerce` section, enable :guilabel:`Follow up abandoned carts`.
#. Set the time delay after which the reminder email is sent in the :guilabel:`Send after` field,
   then click :guilabel:`Save`.
#. Click the :icon:`fa-arrow-right` :guilabel:`Customize Abandoned Email Template`
   link to configure an :doc:`email template </applications/general/companies/email_template>`.

To view all abandoned carts,  go to :menuselection:`Website --> eCommerce --> Orders` and remove
the :guilabel:`Confirmed` filter. Open the :icon:`fa-caret-down` dropdown menu and, under
:guilabel:`Filters`, select :guilabel:`Abandoned`.

.. note::
   - To receive reminders, customers must have entered their contact details during
     checkout or be logged in when adding items to the cart. Without the customer details, it is

>      just a :ref:`quotation <ecommerce/handling/sales>` with no follow-up.

Reviewer: ```suggestion
     checkout or be logged in when adding items to the cart. Without the customer details, it remains a :ref:`quotation <ecommerce/handling/sales>` with no follow-up.
```
(but great idea to phrase it this way!)

   confirmed.

.. seealso::
   :doc:`shipping`

.. _ecommerce/handling/returns:

Returns and refunds
-------------------

Customers can :doc:`return </applications/sales/sales/products_prices/returns>` products
through the :doc:`customer portal </applications/general/users/user_portals>` by navigating to the
:guilabel:`Your Orders` section, selecting the relevant order, and clicking the :guilabel:`Return`
button. A return document is automatically generated.

.. image:: order_handling/return-button.png
   :alt: Return button in the portal.

Once the returned items are received, initiate the :doc:`return and refund process
</applications/sales/sales/products_prices/returns>` in the backend.

.. note::
   - :doc:`Returning </applications/sales/sales/products_prices/returns>` products may not be
     possible depending on the :doc:`product type
     </applications/inventory_and_mrp/inventory/product_management/configure/type>`.
   - To issue :doc:`refunds </applications/finance/accounting/customer_invoices/credit_notes>`,
     a refund-compatible :ref:`payment provider <payment_providers/supported_providers>` needs to

>      be enabled.

Reviewer: I think it not only needs to be enabled, but it must also have been used by the customer to complete the payment so i would rephrase this:

```suggestion
   - To issue :doc:`refunds </applications/finance/accounting/customer_invoices/credit_notes>`, the payment must have been made through a :ref:`refund-compatible payment provider <payment_providers/supported_providers>`.
```
